tests/wicked/advanced_sut.pm

```diff
@@ -21,11 +22,27 @@ use utils 'systemctl';
 use lockapi;
 use mmapi;
 
+sub get_test_result {
+    my ($self, $ip, $ip_version) = @_;
+    my $timeout = "60";
+    my $ret = $self->ping_with_timeout(ip => "$ip", timeout => "$timeout", ip_version => $ip_version);
+    if (!$ret) {
+        record_info("Can't ping IP $ip", result => 'fail');
+        return "FAILED";
+    }
+    else {
+        return "PASSED";
+    }
+}
+
 sub run {
     my ($self) = @_;
-    record_info('Test 1', 'Create a gre interface from legacy ifcfg files');
+    my %results;
+    my $iface      = script_output('ls /sys/class/net/ | grep -v lo | head -1');
     my $gre_config = '/etc/sysconfig/network/ifcfg-gre1';
+    my $sit_config = '/etc/sysconfig/network/ifcfg-sit1';
```

Comment:
I would also move this line to the corresponding test. If we keep all the things that correspond to a sub-test in the sub-test block, we can think of a smart way of grouping things together to not write similar lines again and again.